packages/session-replay-browser/src/events/events-idb-store.ts

```diff
@@ -0,0 +1,308 @@
+    await this.transitionFromKeyValStore(sessionId);
+  }
+
+  /**
+   * Determines whether to send the events list to the backend and start a new
+   * empty events list, based on the size of the list as well as the last time sent
+   * @param nextEventString
+   * @returns boolean
+   */
+  shouldSplitEventsList = (events: Events, nextEventString: string): boolean => {
+    const sizeOfNextEvent = new Blob([nextEventString]).size;
+    const sizeOfEventsList = new Blob(events).size;
+    if (sizeOfEventsList + sizeOfNextEvent >= this.maxPersistedEventsSize) {
+      return true;
+    }
+    if (this.timeAtLastSplit !== null && Date.now() - this.timeAtLastSplit > this.interval && events.length) {
+      this.interval = Math.min(MAX_INTERVAL, this.interval + MIN_INTERVAL);
+      this.timeAtLastSplit = Date.now();
+      return true;
+    }
+    return false;
+  };
+
+  getSequencesToSend = async () => {
+    try {
+      const sequencesToSend = (await this.db?.getAll<'sequencesToSend'>(
+        sequencesToSendKey,
+      )) as SendingSequencesIDBReturn[];
+
+      return sequencesToSend;
+    } catch (e) {
+      this.loggerProvider.warn(`${STORAGE_FAILURE}: ${e as string}`);
+    }
+    return undefined;
+  };
+
+  storeCurrentSequence = async (sessionId: number) => {
+    try {
+      if (!this.db) {
+        return undefined;
+      }
+      const currentSequenceData = await this.db.get<'sessionCurrentSequence'>(currentSequenceKey, sessionId);
+      if (!currentSequenceData) {
+        return undefined;
+      }
+
+      const sequenceId = await this.db.put<'sequencesToSend'>(sequencesToSendKey, {
+        sessionId: sessionId,
+        events: currentSequenceData.events,
+      });
+
+      await this.db.put<'sessionCurrentSequence'>(currentSequenceKey, { sessionId, events: [] });
+
+      return {
+        ...currentSequenceData,
+        sessionId,
+        sequenceId,
+      };
+    } catch (e) {
+      this.loggerProvider.warn(`${STORAGE_FAILURE}: ${e as string}`);
+    }
+    return undefined;
+  };
+
+  addEventToCurrentSequence = async (sessionId: number, event: string) => {
+    try {
+      const tx = this.db?.transaction<'sessionCurrentSequence', 'readwrite'>(currentSequenceKey, 'readwrite');
+      if (!tx) {
+        return;
+      }
+      const sequenceEvents = await tx.store.get(sessionId);
+      if (!sequenceEvents) {
+        await tx.store.put({ sessionId, events: [event] });
+        return;
+      }
+      let eventsToSend;
+      if (this.shouldSplitEventsList(sequenceEvents.events, event)) {
+        eventsToSend = sequenceEvents.events;
+        // set store to empty array
+        await tx.store.put({ sessionId, events: [event] });
+      } else {
+        // add event to array
+        const updatedEvents = sequenceEvents.events.concat(event);
+        await tx.store.put({ sessionId, events: updatedEvents });
+      }
```

Comment:
I think this should be fine for now, but I'm curious how well it scales long term. We may need a strategy where we need to write to multiple rows.

Reply:
Like we need more than one current sequence? Or what do you mean by multiple rows?

Reply:
I was thinking if we're doing a constant grab the array, add to it, then put it back there may be contention on the key. The idea was to potentially split the array across multiple rows since there's a bit of overhead in adding to that array.